app/url/create-url/createUrlInteractor.js

```diff
@@ -41,16 +43,12 @@ class CreateUrlInteractor {
             return;
         }
 
-        const hash = this.jwt.sign(
-            {
-                pasteId: request.pasteId,
-                visibility: paste._visibility
-            },
-            process.env.SECRET_KEY,
-            {
-                expiresIn: DEFAULT_TOKEN_EXPIRES_AFTER_HOURS
-            }
-        );
+        const hash = await this.authTokenService.sign({ pasteId: request.pasteId, visibility: paste.getVisibility() });
+
+        if (paste.getVisibility() === visibility.PRIVATE) {
```

Comment:
Same. Replace this comparison with single method. `paste.isPrivate()` or `isProtected()` or smth similar